test/components/hasnot_test.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:oxygen/oxygen.dart';
+import 'package:test/test.dart';
+
+class ComponentA extends ValueComponent<void> {}
+
+class ComponentB extends ValueComponent<void> {}
+
+class InitSystem extends System {
+  late final Query initQuery;
+  late final Query query;
+  @override
+  void init() {
+    initQuery = createQuery([Has<ComponentA>(), HasNot<ComponentB>()]);
+    query = createQuery([Has<ComponentA>(), Has<ComponentB>()]);
+  }
+
+  @override
+  void execute(double delta) {
+    for (final entity in initQuery.entities) {
+      entity.add<ComponentB, void>();
+    }
+  }
+}
+
+void main() {
```

Comment:
Some new lines between assertion of expectations might be nice as well, for readability sake ;-)